operations/image_management/Import_External_Image_to_IMS.md

```diff
@@ -23,217 +36,216 @@ the following procedure can be used to add this external image to IMS and upload
+[Create image manifest file and upload to S3](#6-create-image-manifest-file-and-upload-to-s3).
 
-    1. If the above variable is not set, then set `IMS_INITRD_FILENAME` to the file name of the initrd file
-       to be uploaded.
+1. Navigate to the directory containing the artifact files.
 
-        ```bash
-        IMS_INITRD_FILENAME=initrd
-        ```
+1. (`ncn-mw#`) Verify that all image artifacts exist in the current working directory.
 
-    1. Navigate to the directory containing the initrd file. Verify that it exists in your current working
-       directory.
+    > If necessary, modify the following command to reflect the actual set of artifacts included in the image.
+
+    ```bash
+    ls -al "${IMS_ROOTFS_FILENAME}" "${IMS_INITRD_FILENAME}" "${IMS_KERNEL_FILENAME}"
+    ```
+
+1. (`ncn-mw#`) Upload the artifacts to S3.
+
+    1. Upload the SquashFS image root to S3.
 
         ```bash
-        ls $IMS_INITRD_FILENAME
+        cray artifacts create boot-images "${IMS_IMAGE_ID}/${IMS_ROOTFS_FILENAME}" "${IMS_ROOTFS_FILENAME}"
         ```
 
-    1. Record the `md5sum` of the initrd to be uploaded.
+    1. Upload the kernel to S3.
 
         ```bash
-        IMS_INITRD_MD5SUM=$(md5sum $IMS_INITRD_FILENAME | awk '{ print $1 }')
+        cray artifacts create boot-images "${IMS_IMAGE_ID}/${IMS_KERNEL_FILENAME}" "${IMS_KERNEL_FILENAME}"
         ```
 
-    1. Upload the initrd to S3.
+    1. Upload the `initrd` to S3.
 
         ```bash
-        cray artifacts create boot-images $IMS_IMAGE_ID/$IMS_INITRD_FILENAME $IMS_INITRD_FILENAME
+        cray artifacts create boot-images "${IMS_IMAGE_ID}/${IMS_INITRD_FILENAME}" "${IMS_INITRD_FILENAME}"
         ```
 
-### Image manifest
+### 6. Create, upload, and register image manifest
 
-1. Create an image manifest and upload it to S3.
+HPE Cray uses a manifest file that associates multiple related boot artifacts \(kernel, `initrd`, image root, etc.\) into
+an image description that is used by IMS and other services to boot nodes. Artifacts listed within the manifest are
+identified by a `type` value:
 
-    HPE Cray uses a manifest file that associates multiple related boot artifacts \(kernel, `initrd`, `rootfs`, etc.\) into
-    an image description that is used by IMS and other services to boot nodes. Artifacts listed within the manifest are
-    identified by a `type` value:
+* `application/vnd.cray.image.rootfs.squashfs`
+* `application/vnd.cray.image.initrd`
+* `application/vnd.cray.image.kernel`
+* `application/vnd.cray.image.parameters.boot`
 
-    * `application/vnd.cray.image.rootfs.squashfs`
-    * `application/vnd.cray.image.initrd`
-    * `application/vnd.cray.image.kernel`
-    * `application/vnd.cray.image.parameters.boot`
+1. (`ncn-mw#`) Generate an image manifest file.
 
-    1. Generate an image manifest file.
+    > If necessary, modify the following example to reflect the actual set of artifacts included in the image.
```

Comment:
Oh, you mean you didn't use the cat command provided, but wrote the file otherwise and just copied in the contents? Got it. I can add a note about that.